bundle/edu.gemini.obslog/src/main/java/edu/gemini/obslog/instruments/GNIRSLogSegment.java

```diff
@@ -35,40 +30,34 @@
     private static final String READMODE_KEY = "readMode";
     private static final String COADDS_KEY = "coadds";
     private static final String EXPOSURETIME_KEY = "exposureTime";
-    private static final String WOLLASTON_KEY = "wollaston";
     private static final String PIXELSCALE_KEY = "pixelScale";
     private static final String CAMERA_KEY = "camera";
 
-    public GNIRSLogSegment(List<OlLogItem> logItems, OlLogOptions obsLogOptions) {
+    public GNIRSLogSegment(final List<OlLogItem> logItems, final OlLogOptions obsLogOptions) {
         super(SEG_TYPE, logItems, obsLogOptions);
     }
 
-    private void _decorateFilter(ConfigMap map) {
+    private void _decorateFilter(final ConfigMap map) {
         if (map == null) return;
 
         // Note that filterValue can return null here if there is no filter in the sequence.  Then this code needs
         // to divine it from other values based upon SeqExec details
         String filterValue = map.sget(FILTER_KEY);
 
-        String typeValue = map.sget(TYPE_KEY);
-        if (typeValue == null) typeValue = EMPTY_STRING;
-
         // This should work if filterValue == null - the check is added to keep
         // oldValueOf from complaining
         if (filterValue != null) {
-            GNIRSParams.Filter f = GNIRSParams.Filter.getFilter(filterValue, null);
+            final GNIRSParams.Filter f = GNIRSParams.Filter.getFilter(filterValue, null);
             if (f != null) {
+                final String typeValue = ImOption.apply(map.sget(TYPE_KEY)).getOrElse("");
                 map.put(FILTER_KEY, !typeValue.equals("DARK") ? f.logValue() : "DARK");
```

Comment:
Okay, I rewrote the class in Scala and think it's much improved, but it requires testing.